plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -55,46 +57,49 @@
     public static final String FILE_MODIFIED_TIME_COLUMN_NAME = "$file_modified_time";
     public static final Type FILE_MODIFIED_TIME_TYPE = TIMESTAMP_TZ_MILLIS;
 
-    private final String name;
-    private final Type type;
+    private final String baseColumnName;
+    private final Type baseType;
     private final OptionalInt fieldId;
     // Hold field names in Parquet files
     // The value is same as 'name' when the column mapping mode is none
     // The value is same as 'delta.columnMapping.physicalName' when the column mapping mode is id or name. e.g. col-6707cc9e-f3aa-4e6b-b8ef-1b03d3475680
-    private final String physicalName;
+    private final String basePhysicalColumnName;
     // Hold type in Parquet files
     // The value is same as 'type' when the column mapping mode is none
     // The value is same as 'delta.columnMapping.physicalName' when the column mapping mode is id or name. e.g. row(col-5924c8b3-04cf-4146-abb5-2c229e7ff708 integer)
-    private final Type physicalType;
+    private final Type basePhysicalType;
     private final DeltaLakeColumnType columnType;
+    private final Optional<DeltaLakeColumnProjectionInfo> projectionInfo;
 
     @JsonCreator
     public DeltaLakeColumnHandle(
-            @JsonProperty("name") String name,
-            @JsonProperty("type") Type type,
+            @JsonProperty("baseColumnName") String baseColumnName,
+            @JsonProperty("baseType") Type baseType,
             @JsonProperty("fieldId") OptionalInt fieldId,
```

Comment:
is `fieldId` base column field id, or leaf field id?
(i hope it's leaf, please confirm)

Reply:
It is the base column field id - see https://github.com/trinodb/trino/blob/314b19ca6e66026994ca90c99339ef27b34d6ccd/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java#L353-L366

We currently don't have support for leaf field ID (nested field ids in the `DeltaLakeColumnMetadata`).

Reply:
Do note that the field ID is not used in the Parquet reading via Hive mechanism:

https://github.com/trinodb/trino/blob/314b19ca6e66026994ca90c99339ef27b34d6ccd/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java#L160-L170

Reply:
Maybe we should lookup after `fieldId` to be on the safe side 

https://github.com/trinodb/trino/blob/314b19ca6e66026994ca90c99339ef27b34d6ccd/plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java#L473

Reply:
Created follow-up https://github.com/trinodb/trino/issues/17247

Reply:
We currently use the `fieldId` to retrieve the appropriate `baseColumnName` for the `HiveColumnHandle`:

https://github.com/trinodb/trino/blob/47d15d0c37ff8f9266f81269e562242caf0db342/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java#L265-L279

The field id to name mapping is currently built out of the base fields:

https://github.com/trinodb/trino/blob/47d15d0c37ff8f9266f81269e562242caf0db342/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java#L229-L243

Note that the method above builds the id to name mapping only for the base fields and does not take into account that group fields could have nested fields.

While building the `DeltaLakeColumnHandle` in the `DeltaLakeSchemaSupport` we only read the base `fieldId` - we should be eventually reading the entire tree of field ids in the type.